Update to GOV.UK Frontend

nickcolley opened the pull request:
This is an attempt to do a cleaner merge of the changes from the https://github.com/alphagov/govuk-prototype-kit-private-beta.

Includes:

- Removal of Elements, Template, Toolkit
- Migration of all code
- Updates to guidance relating to Elements, Template, Toolkit
- New cookie banner component

If you want to review all changes without  whitespace: https://github.com/alphagov/govuk_prototype_kit/pull/512/files?w=1

The other parts that are missing are being tracked with this milestone: https://github.com/alphagov/govuk_prototype_kit/milestone/1

https://trello.com/c/SrxrezjZ/1061-add-govuk-frontend-to-the-govuk-prototype-kit

igloosi commented on app/assets/sass/unbranded.scss:
```diff
@@ -1,30 +1,12 @@
-.unbranded {
-  background: $white;
+// Remove canvas background colour, as is used with the GOV.UK Footer.
+$govuk-canvas-background-colour: $govuk-body-background-colour;
```
no need for this as we're not including the footer

nickcolley replied:
If we don't do this we get a grey box at the bottom of the screen, since the `<body>` does not extend the same length as `<html>`

igloosi commented on server.js:
```diff
@@ -190,6 +205,7 @@ app.get('/prototype-admin/download-latest', function (req, res) {
 if (useDocumentation) {
   // Copy app locals to documentation app locals
   documentationApp.locals = app.locals
+  documentationApp.locals.serviceName = 'Prototype Kit'
```
this needs to be changed to how we've amended it in beta
  documentationApp.locals = Object.assign({}, app.locals)
  documentationApp.locals.serviceName = 'Prototype Kit'

joelanman replied:
dunno if its github being weird, but the line looks how you suggested already?
`documentationApp.locals = Object.assign({}, app.locals)`

joelanman replied:
ah it's because you commented on line 199 instead of 198 :)

joelanman commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "express": "4.15.2",
     "express-session": "^1.13.0",
     "express-writer": "0.0.4",
+    "govuk-frontend": "0.0.32",
```
should this be `1.0.0`?

joelanman commented on .gitignore:
```diff
@@ -5,7 +5,6 @@
 .port.tmp
 public
 lib/govuk_template.html
-govuk_modules
```
good catch! This isn't in the beta work

joelanman commented on app/assets/sass/examples.scss:
```diff
@@ -1,22 +0,0 @@
-@import "_colours";
```
another good catch! This file has been hanging around for ages

joelanman commented on docs/documentation/writing-css.md:
```diff
@@ -18,23 +18,12 @@ Every time a change happens in [application.scss](../app/assets/sass/application
 Try starting the app and adding some styles to `application.scss`. If you open `application.css` you should now see the compiled version of those styles.
 
-## Using the govuk_frontend_toolkit
+## Using the GOV.UK Frontend
```
delete 'the'

joelanman replied:
to be honest its probably best to remove this section, it's not very useful and tells people to do something thats not very clear, or we already do for them?

joelanman commented on docs/documentation/writing-css.md:
```diff
@@ -18,23 +18,12 @@ Every time a change happens in [application.scss](../app/assets/sass/application
+Imports are done from either [GOV.UK Frontend](https://github.com/alphagov/govuk-frontend/tree/master/src/) folder or the [app/assets/sass](../app/assets/sass) folder application.scss sits in. The latter means you can create your own partial files to import.
 
-Note that the convention is to start the name of any partial with an underscore, like those in the toolkit.
+Note that the convention is to start the name of any partial with an underscore, like those in the GOV.UK Frontend.
```
like those in ~the~ GOV.UK Frontend

joelanman commented on docs/documentation/writing-css.md:
```diff
@@ -18,23 +18,12 @@ Every time a change happens in [application.scss](../app/assets/sass/application
-      @include bold-48;
-
-The line `@import '_typography';` makes all the code in [_typography.scss](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/stylesheets/_typography.scss) available. The `h1` can therefore be styled in the 48pt bold form of the font by using `@include bold-48;` to call the `bold-48` mixin.
+You can use the Sass libraries in the [govuk-frontend](https://github.com/alphagov/govuk-frontend) by importing the files from there directly into `application.scss`.
```
in ~the~ govuk-frontend

joelanman commented on app/views/index.html:
```diff
@@ -25,36 +25,20 @@ <h2 class="govuk-heading-m">About this page</h2>
+      <ul class="govuk-list govuk-list--bullet">
+        <li><a href="/docs/tutorials-and-examples">Example pages and documentation</a></li>
+        <li>
+          <a href="https://govuk-design-system-production.cloudapps.digital">
```
https://design-system.gov.uk

joelanman commented on docs/views/tutorials-and-examples.html:
```diff
@@ -163,10 +163,10 @@ <h2 class="govuk-heading-m">Page templates</h2>
-      <p class="text">
-        <a href="https://govuk-elements.herokuapp.com/">GOV.UK Elements</a> is a guide to making your prototype look like a GOV.UK
+      <p>
+        <a href="https://govuk-design-system-production.cloudapps.digital">GOV.UK Design System</a> is a guide to making your prototype look like a GOV.UK
```
https://design-system.gov.uk

joelanman commented on docs/views/cookies.html:
```diff
@@ -42,26 +42,23 @@ <h2 class="govuk-heading-l">Our cookie message</h2>
 
       <p>You will see a message about cookies when you first visit the GOV.UK Prototype Kit. We’ll store a cookie so that your computer knows you’ve seen it and knows not to show it again.</p>
```
does this fix a problem with tables? Or just move it to using a macro?

joelanman commented on docs/documentation/tips-and-tricks.md:
```diff
@@ -15,30 +15,54 @@ You can change the service name by editing the file '/app/config.js'.
-      </ul>
-      -->
-    </nav>
+Import the header component macro place it in the `{% block header %}`and provide `navigation` items as shown below.
```
does this work fix something or just move it to macros? If its just about macros can we skip this commit, as we're not ready to push macros in prototyping

joelanman commented on package.json:
```diff
@@ -46,5 +46,8 @@
     ]
+  },
+  "devDependencies": {
+    "cookie-parser": "^1.4.3"
```
should this be dependencies, not devDependencies? Is this a different approach to the one taken in the beta repo?

nickcolley replied:
I think the dependencies for the kits is a bit of a mess, I'll move this into dependencies for now but they're definitely not all regular dependencies...

joelanman commented on docs/views/examples/confirmation-page.html:
```diff
@@ -9,15 +9,14 @@
-          <strong class="bold">HDJ2123F</strong>
-        </p>
-      </div>
+      {% set panelHtml %}
```
can we revert to not use macro?

nickcolley commented:
@joelanman 

- changed all examples that users will interact with to avoid macros
- squashed the commits you mentioned
- fixed an issue where there was a stray deleted file in a commit it didnt belong.

joelanman commented:
Sorry I meant small changes plus reverting macros, not that that was small, I wasn't very clear.

joelanman commented:
Wondering if the Cookie banner work would be better as a separate PR? Is it part of 'Update to govuk frontend'?

nickcolley commented:
@joelanman yeah, I think it makes sense since otherwise this PR would break cookie support.
